docs/src/Groups/subgroups.md

```diff
@@ -43,17 +43,13 @@ intersect(::T, V::T...) where T<:GAPGroup
 The following functions return a vector of subgroups.
 
 ```@docs
-subgroups(G::GAPGroup)
```

Comment:
(I thought I'd already left this as a comment, but now it's not here -- maybe I once again forgot to press "submit"? Or maybe I left this as a comment elsewhere (and you perhaps even answered it there) -- sorry in that case, and please point me to it.)

Why is this dropping `subgroups` altogether? Isn't the plan to provide both `*subgroups` and `*subgroup_classes` commands in parallel for all four commands?

Reply:
Currently the proposed changes deprecate `maximal_subgroups`; there were no `complements` and `hall_subgroups` before (just a very old, already deprecated `hall_subgroup`); and `subgroups` will remain for the methods that come from Hecke (not for `GAPGroup`s), with a different behaviour which we did not want to touch.

Reply:
Yes, I think `subgroups` should exist for the casual user and it should return subgroups. Not sure if the `Iterators.flatten(subgroup_classes(G))` does this.

Reply:
O.k., we define and document `subgroups` for a `GAPGroup`, and the result is an iterator of subgroups (without embeddings), analogous to `subgroups(abelian_group([2,3,4]))`, which returns a `Hecke.SubgroupIterator`.
The docstring will contain a hint that probably calling `subgroup_classes` would be better.

Should `complements`, `hall_subgroups`, `low_index_subgroups`, `maximal_subgroups` be treated analogously?

Reply:
Yes